### PR TITLE
PXP-466: [Wukong CLI] Refactor the configuration logic

### PR DESCRIPTION
**Pull request by jk-gan**
## Summary

Previously, when the config file format is updated with breaking changes, all the `wukong` commands are not able to run. The user have to manually remove the old config file and run `wukong init` to re-initialize. 

This is because we load the config before we execute the command, hence, when the config file format is different, the cli will panic even when we run `wukong --help` and `wukong --version`.

With this PR, we only read the config when necessary. As the result, user can now run `wukong init` to re-initialize it without manually remove the old config file. 1 step saving!

Ticket: [PXP-466](https://mindvalley.atlassian.net/browse/PXP-466)

### Added
- [x] introduce 2 env variables for testing purpose:
    - `WUKONG_DEV_CONFIG_FILE` -> to customize the config file path, this is useful to use customize config file during test
    - `WUKONG_DEV_TIMEZONE` -> to customize the timezone used to format the datetime
- [x] add integration tests for all the commands except the interactive one, need to find a way to test it in the future

### Changed
- [x] update dependencies
- [x] use `CliError` to handle the uninitialized and unauthenticated situation whenever it is needed

### Fixed
- [x] `wukong init` now can be used when the config file broken

[PXP-466]: https://mindvalley.atlassian.net/browse/PXP-466?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/60#" title="Author self-approved">jk-gan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/main.rs
Comment (onimsha) [inline src/main.rs]: should remove commented code.

File: src/main.rs
Comment (onimsha) [inline src/main.rs]: same here

**Review by onimsha — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/60#" title="Author self-approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/60#pullrequestreview-1304985435" title="Approved">onimsha</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
